hack/newTargetGen/main.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2021 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+)
+
+type component struct {
+	Name          string
+	UppercaseName string
+	FullCaps      string
+}
+
+func (a *component) replaceTemplates(filename, outputname string) {
+	tmp1, err := template.ParseFiles(filename)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	file, err := os.Create(outputname)
+	if err != nil {
+		log.Fatalf("failed creating file: %s", err)
+	}
+	defer file.Close()
+
+	err = tmp1.Execute(file, a)
+
+	if err != nil {
+		fmt.Println(err)
+	}
+
+}
+
+func main() {
+	temp := &component{}
+	// var capsName string
+
+	fmt.Print("Enter the target name: ")
+	fmt.Scanf("%s", &temp.UppercaseName)
+
+	temp.Name = strings.ToLower(temp.UppercaseName)
+	temp.FullCaps = strings.ToUpper(temp.UppercaseName)
+	// fmt.Print("Enter the ALL CAPS VERISON of the target name: ")
+	// fmt.Scanf("%s", &capsName)
+
+	// TODO add naming validation here
+
+	// make cmd folder
+	path := fmt.Sprintf("cmd/%s", temp.Name+"-adapter")
```

Comment:
`Sprintf` feels a bit like an overkill for a string concatenation.
The same result can be achieved without any function call, and passed directory to your path manipulation below: `"cmd/" + temp.Name+"-adapter"`.